src/text-to-speech-adapter.js

```diff
@@ -14,6 +14,8 @@ function textToSpeechAdapter(conf) {
   return function(chatbot) {
     let ttsMessagesQueue = [];
     let speaking = false;
+    let finished = true;
```

Comment:
When a single reply contains many paragraphs the `speaking` variable changes state between the paragraphs and not when the whole reply finishes.

This is the problem, so in order for us to know if the entire reply is read, we need to add a `finished` variable to check if all the text of a single reply is finished being spoken. If the text has finished being spoken, we can continue on the next reply text.